### PR TITLE
Apply party type to their-details and check-and-send CCJ pages

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link ###

https://tools.hmcts.net/jira/browse/ROC-2044

### Change description ###

Adds more party details to the 'their details page' depending on party type + displays party details on the check & send page.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[n] No
```

### CODE REVIEW

File: src/main/features/ccj/views/check-and-send.njk [hunk 1]
```diff
@@ -11,14 +12,15 @@
     {{ csrfProtection(csrf) }}
 
     {{ tableStart("theirDetails", "Their details") }}
-    {{ row('Full name:', user.claim.claimData.defendant.name) }}
```
Comment (geek-soft) [inline src/main/features/ccj/views/check-and-send.njk]: The only issue I spotted is that 'Change' link is present in 'Name', 'Business name' or 'Contact person' rows. Need to reimplement it a bit.

**Comment by geek-soft** [comment]
Applying party type to CCJ date of birth page will come as separate PR. Due to the agreement DOB should be only displayed to individuals.

File: src/main/features/ccj/views/their-details.njk [hunk 2]
```diff
@@ -17,18 +18,50 @@
...
+              heading: 'Their address',
+              name: undefined
+            },
+            showCorrespondenceAddressSection = false
+          )
+        }}
 
-          <h2 class="heading-medium">{{ t('Email address') }}</h2>
-          <p>{{ claim.claimData.defendant.email }}</p>
+        {{ readOnlyField(label = 'Email address', value = claim.claimData.defendant.email) }}
```
Comment (timja-kainos) [inline src/main/features/ccj/views/their-details.njk]: this is left over from before, but I think there should be a `if defined` wrapped around the email block.
Might be worth checking

File: src/main/features/claim/views/claimant-company-details.njk [hunk 3]
```diff
@@ -12,13 +12,18 @@
...
-            addressHeading = "Company address",
-            addressHint = "Enter the company’s registered office or a company address that has a connection with the claim.",
-            hasCorrespondenceAddress = true
+            fields = [
+              { label: 'Company name', name: 'name' },
+              { label: 'Contact person', name: 'contactPerson' }
+            ],
+            addressSection = {
+              heading: 'Company address',
+              hint: "Enter the company's registered office or a company address that has a connection with the claim.",
```
Comment (timja-kainos) [inline src/main/features/claim/views/claimant-company-details.njk]: company’s

File: src/main/features/claim/views/claimant-organisation-details.njk [hunk 4]
```diff
@@ -12,13 +12,18 @@
...
-            addressHeading = "Organisation address",
-            addressHint = "Enter the organisation's main office or an organisation's address that has a connection with the claim.",
-            hasCorrespondenceAddress = true
+            fields = [
+              { label: 'Organisation name', name: 'name' },
+              { label: 'Contact person', name: 'contactPerson' }
+            ],
+            addressSection = {
+              heading: 'Organisation address',
+              hint: "Enter the organisation's main office or an organisation's address that has a connection with the claim.",
```
Comment (timja-kainos) [inline src/main/features/claim/views/claimant-organisation-details.njk]: organisation’s

File: src/main/features/claim/views/defendant-company-details.njk [hunk 5]
```diff
@@ -12,13 +12,18 @@
...
-            addressHeading = "Company address",
-            addressHint = "Enter the company’s registered office or a company address that has a connection with the claim.",
-            hasCorrespondenceAddress = false
+            fields = [
+              { label: 'Company name', name: 'name' },
+              { label: 'Contact person (optional)', name: 'contactPerson' }
+            ],
+            addressSection = {
+              heading: 'Company address',
+              hint: "Enter the company's registered office or a company address that has a connection with the claim.",
```
Comment (timja-kainos) [inline src/main/features/claim/views/defendant-company-details.njk]: company’s

File: src/main/features/claim/views/defendant-individual-details.njk [hunk 6]
```diff
@@ -12,12 +12,17 @@
...
-            nameLabel = "Full name",
-            addressHeading = "Their residential address",
-            addressHint = "You must enter the defendant's residential address to make a claim against them.",
-            hasCorrespondenceAddress = false
+            fields = [
+              { label: 'Full name', name: 'name' }
+            ],
+            addressSection = {
+              heading: 'Their residential address',
+              hint: "You must enter the defendant's residential address to make a claim against them.",
```
Comment (timja-kainos) [inline src/main/features/claim/views/defendant-individual-details.njk]: defendant’s

File: src/main/features/claim/views/defendant-sole-trader-details.njk [hunk 7]
```diff
@@ -12,12 +12,18 @@
...
-            addressHeading = "Their address",
-            addressHint = "You must enter the defendant's residential or trade address to make a claim against them.",
-            hasCorrespondenceAddress = false
+            fields = [
+              { label: 'Full name', name: 'name' },
+              { label: 'Business name (optional)', name: 'businessName' }
+            ],
+            addressSection = {
+              heading: 'Their address',
+              hint: "You must enter the defendant's residential or trade address to make a claim against them.",
```
Comment (timja-kainos) [inline src/main/features/claim/views/defendant-sole-trader-details.njk]: defendant’s

File: src/main/views/includes/macros/partyDetails.njk [hunk 8]
```diff
@@ -1,61 +1,81 @@
...
+{% from "includes/macros/table.njk" import tableStart,tableEnd, row, rowWithMultipleValue %}
 
-{% macro individualDetailsFragment(form, nameLabel = 'Full name', addressHeading = 'Address', addressHint, hasCorrespondenceAddress = false) %}
-  {{ textInput(t(nameLabel), 'name', form, autocomplete = 'name') }}
+{% macro partyDetailsFragment(form, fields, addressSection, showCorrespondenceAddressSection = false) %}
+  {% for field in fields %}
+    {% if field.readOnly %}
+      {{ readOnlyField(field.label, field.value) }}
+    {% else %}
+      {{ textInput(form = form, label = t(field.label), name = field.name, autocomplete = field.autocomplete) }}
```
Comment (timja-kainos) [inline src/main/views/includes/macros/partyDetails.njk]: I don't think t should be applied here, done inside the textInput instead

File: src/main/views/includes/macros/table.njk [hunk 9]
```diff
@@ -10,23 +10,23 @@
...
+{% macro row( label, value, changeLink ) %}
   <div class="bottom-border">
     <div class="column-one-third">
       <span>{{ t( label ) }}</span>
     </div>
     <div class="column-one-third">
       <span>{{ value }}</span>
     </div>
-    {% if changeLink !== '' %}
+    {% if changeLink !== undefined %}
```
Comment (timja-kainos) [inline src/main/views/includes/macros/table.njk]: `if changeLink` should be fine

File: src/main/views/includes/macros/table.njk [hunk 10]
```diff
@@ -36,7 +36,7 @@
         <span class="block">{{ value }}</span>
        {% endfor %}
     </div>
-    {% if changeLink !== '' %}
+    {% if changeLink !== undefined %}
```
Comment (timja-kainos) [inline src/main/views/includes/macros/table.njk]: same here

File: src/main/views/includes/macros/partyDetails.njk (hunk 8 above)
Comment (geek-soft) [reply src/main/views/includes/macros/partyDetails.njk]: A `input` does not translate labels itself. Something we should cleanup service wide, did not want to touch this now.

File: src/main/features/ccj/views/their-details.njk (hunk 2 above)
Comment (geek-soft) [reply src/main/features/ccj/views/their-details.njk]: Good reminder, confirmed it with @tarik-johnston1, made it conditional. Will update @pragnesh-patel on this.

**Comment by geek-soft** [comment]
Demoed to business, all looks alright.

**Pull request merged**